### ISSUE TITLE
Flush IPv4 LinkLocal IP when IPv4 Static IP is assigned 

### ISSUE THREAD

**Issue by raviteja-b** [systemd/systemd#25424]
### systemd version the issue has been seen with

249

### Used distribution

ubuntu 

### Linux kernel version used

5.15.76-233f3dc

### CPU architectures issue was seen on

arm

### Component

systemd-networkd

### Expected behaviour you didn't see

Its expected to flush IPv4 link local IP when Static IP configured on ethernet interface.
set "LinkLocalAddressing=yes"

### Unexpected behaviour you saw

Noticing IPv4 link Local IP even after configuring Static IP

### Steps to reproduce the problem

Configure static IP on ethernet interface, still it has both static IP and link local IP.

**Comment by raviteja-b** [comment]
I noticed that Fallback option was removed which systemd-networkd used to support earlier.
how to remove IPv4 link local IP when IPv4 Static IP configured?

**Comment by yuwata** [comment]
What's the usecase of specifying both static address and an IPv4LL address at the same time??

**Comment by raviteja-b** [comment]
> What's the usecase of specifying both static address and an IPv4LL address at the same time??

yes, we don't have use-case of having ipv4 linklocal IP as soon as Static IP configured

**Comment by yuwata** [comment]
I mean, why do you enable the IPv4LL setting even though you have a static address in the same .network file? Or do you assign an static IPv4 address by using an external tool?

**Comment by raviteja-b** [comment]
> I mean, why do you enable the IPv4LL setting even though you have a static address in the same .network file? Or do you assign an static IPv4 address by using an external tool?

No, By default it configures 'LinkLocalAddressing=yes' irrespective of DHCP/Static IP.
by the way, I have tried configuring 'LinkLocalAddressing=ipv4-fallback' and it works as expected.
but I see fallback option removed from systemd-networkd documentation

**Comment by raviteja-b** [comment]
@yuwata As part of RFE, will make changes to provide same Link Local IPv4 behaviour irrespective of DHCP/Static IP configuration?
Please capture these Linklocal IPv4 behaviour changes in systemd-networkd documentation as well.

**Comment by sunharis** [comment]
Hi @yuwata , I see you have started work on this. Can you please let me know if you have any estimation on the time when this change will be available at v254 ? Can it not make it to Jan 2023 release v253 ?

**Comment by yuwata** [comment]
Sorry, I have not started to work on this yet. I'd like to support this on v254, but not sure. There are many things in my TODO list...

**Comment by sunharis** [comment]
> Sorry, I have not started to work on this yet. I'd like to support this on v254, but not sure. There are many things in my TODO list...

Thats fine. Thank you. Do you have an estimated date for this v254 release ?

**Comment by yuwata** [comment]
> > Sorry, I have not started to work on this yet. I'd like to support this on v254, but not sure. There are many things in my TODO list...
> 
> Thats fine. Thank you. Do you have an estimated date for this v254 release ?

Unfortunately, we have no estimate about v254 yet. Sometimes, 3 months after the previous, but sometimes 9 months... I cannot say anything about the future releases.

**Comment by sunharis** [comment]
> > > Sorry, I have not started to work on this yet. I'd like to support this on v254, but not sure. There are many things in my TODO list...
> > 
> > 
> > Thats fine. Thank you. Do you have an estimated date for this v254 release ?
> 
> Unfortunately, we have no estimate about v254 yet. Sometimes, 3 months after the previous, but sometimes 9 months... I cannot say anything about the future releases.

Any plans on this @yuwata ?

**Comment by Bartel-C8** [comment]
Same request here:

I have a setup where I always want a IPv6 LL address.

And the IPv4 should be configurable:

1. DHCP with fallback ipv4 LL address
2. Static (with no ipv4 LL)

I am changing this at runtime (based on config / user input). 
Now the only way is to keep the LinkLocalAddressing inline with the mode above:

1. DHCP=ipv4 , LinkLocalAddressing=yes
2. DHCP=no , LinkLocalAddressing=ipv6, Address=....


Right?

**Comment by raviteja-b** [comment]
@yuwata can you please help look into this issue and see if system-networkd can update fallback option and remove link local IP when static IP is configured

**Comment by raviteja-b** [comment]
@yuwata can you please let us know if there is any workaround to delete link local ip addresses after static ip address is assigned?
so that from application side we can make changes to remove link local IP when static ip address assigned for now

**Comment by raviteja-b** [comment]
I think systemd-networkd needs to add this fallback option back or delete IPv4  link local IP automatically after Static IP assigned
```
By the way, I have tried configuring 'LinkLocalAddressing=ipv4-fallback' and it removes IPv4 Linklocal IP on acquiring Static IP and gets IPv4 Linklocal IP back as soon as static IP gets deleted.
but I see ipv4-fallback option removed from systemd-networkd documentation
```

why it was removed?
